### DISCUSSION TITLE
Allow hidden Navbar & Aside in AppShell, not based on MediaQuery or Breakpoint

### DISCUSSION THREAD

**Discussion by tinybitsofcode** [mantinedev/mantine#3361] [Feature requests]
Hello, I've just started using Mantine and I'm really liking it so far.

I'm trying to create a sort of 'always on' & 'dismissable' NavBar and Aside components using AppShell, following the documentation. It works fine to hide the components based on screen size, but I would be interested to show/hide the components based on my logic and not only screen size.

I'm also using `Portal` to render content dynamically inside, thus hiding the Navbar/Aside programmatically based on a boolean is not a solution (I need the nodes in DOM for the Portal to be active.)

The solution would be to apply a `display: none` style based on a condition, but that doesn't update the main content padding, because the CSS variables for width (navbar or aside width) still stay the same. 

Any ideas on what approach I could take?

### Navbar & Aside visible. The main content is in the middle.
<img width="813" alt="image" src="https://user-images.githubusercontent.com/1685180/212371860-182440a0-677e-4132-bd35-5dd9370d042b.png">

### Navbar & Aside hidden (`display: none`). The main content does not fill up the aside space.
<img width="813" alt="image" src="https://user-images.githubusercontent.com/1685180/212371746-e4fa20f8-41c3-4685-adde-0e95f2d5939a.png">

Thanks in advance :)

**Comment by rtivital** [comment]
You can implement it on your side simply by not rendering the component – https://codesandbox.io/s/musing-mendeleev-hk9qbl?file=/src/App.tsx

**Reply by tinybitsofcode** [reply]
Hello @rtivital. Thank you for your suggestion - unfortunately that won't work since inside my 'Navbar' component I have a Portal target which will break some routes that try to render content inside the Navbar.

It seems the only working solution would be using the logic of 'hiddenForBreakpoint' but not depending on the screen size (somehow)...

Edit: managed to do a workaround by setting the 'width' to 0 whenever not open - this way the padding won't be added. The large 'hidden breakpoint' allows for this functionality to work on any screen size (hacky but yeah). 

https://codesandbox.io/s/wonderful-shape-n57ldc?file=/src/App.tsx
